src/transformers/models/vit/modeling_vit.py

```diff
@@ -116,8 +97,10 @@ def interpolate_pos_encoding(self, embeddings: torch.Tensor, height: int, width:
         # see discussion at https://github.com/facebookresearch/dino/issues/8
         h0, w0 = h0 + 0.1, w0 + 0.1
         patch_pos_embed = nn.functional.interpolate(
-            patch_pos_embed.reshape(1, int(math.sqrt(N)), int(math.sqrt(N)), dim).permute(0, 3, 1, 2),
-            scale_factor=(h0 / math.sqrt(N), w0 / math.sqrt(N)),
+            patch_pos_embed.reshape(1, int(math.sqrt(num_positions)), int(math.sqrt(num_positions)), dim).permute(
+                0, 3, 1, 2
+            ),
```

Comment:
Might be worth putting this on its own line since the formatting makes it hard to read.